src/WebJobs.Script/ScriptConstants.cs

```diff
@@ -248,6 +249,7 @@ public static class ScriptConstants
         public static readonly long DefaultMaxRequestBodySize = 104857600;
 
         public static readonly ImmutableArray<string> SystemLogCategoryPrefixes = ImmutableArray.Create("Microsoft.Azure.WebJobs.", "Function.", "Worker.", "Host.");
+        public static readonly ImmutableHashSet<string> HostCapabilities = ImmutableHashSet.Create<string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
This is immutable, `readonly`, and has no values. It will never contain a value. What is the purpose of it?

Reply:
Today, we don't have any capability that requires negotiation. But if in future, we introduce a capability then we can add in this structure?

For example, there is a new feature "feature1" and during worker probing we want to make sure that host supports "feature1" then we can add as follows in that Host version.

```csharp
HostCapabilities = ImmutableHashSet.Create<string>("feature1",  "feature2");
```

Reply:
That sounds like an `IReadOnlySet<T>` is sufficient. No need for immutable.